include/vptr/virtual_ptr.hpp

```diff
@@ -398,6 +398,13 @@ class PointerMapper {
     }
   }
 
+  /* release_pointer.
+   * Removes the given pointer from the map and do not try to re-use it later.
+   */
+  void release_pointer(const virtual_pointer_t ptr) {
+     m_pointerMap.erase(this->get_node(ptr));
+  }
```

Comment:
Is not clear to me the difference between release and remove. In fact, for a release I would expect that the buffer is released from the pointer map and returned to the user for their own purposes (like std::unique_ptr::release).
Wouldn't it be better if  we put this functionality on the existing remove_pointer and simply pass a flag to it to not re-use?

Reply:
Hum so I don't need the exact same thing as the unique_pre::release. That's because I assume the user already has the buffer.
Anyway I'm fine with adding a flag to remove_pointer to not re-use it. Are you ok with a templated bool parameter? Or do we want a runtime flag for some reason? I think a runtime flag would make sense if we could know whether a buffer is a sub-buffer at runtime but I don't think it's possible. So I would suggest for the template parameter.

Reply:
About the tests that's a fair point. I could work on adding a simple one tomorrow maybe.